app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt

```diff
@@ -393,6 +393,12 @@ class UploadWorker(
                                 makeWikiDataEdit(uploadResult, contribution)
                             }
                             showSuccessNotification(contribution)
+                            if (appContext.contentResolver.persistedUriPermissions.any {
+                                it.uri == contribution.contentUri }) {
+                                appContext.contentResolver.releasePersistableUriPermission(
```

Comment:
Max persistence limit is 512, so clean up is necessary after a successful upload. Ref: https://android.googlesource.com/platform/frameworks/base/+/master/services/core/java/com/android/server/uri/UriGrantsManagerService.java#126